### PR TITLE
Set max verification gas for state dependant signatures

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2319 https://github.com/neo-project/neo/pull/2320

### CODE REVIEW

**Comment by erikzhang** [comment]
Why do you need this?

**Comment by shargon** [comment]
> Why do you need this?

I updated the description, it was sent by @red4sec with the preview5 release.

**Comment by roman-khimov** [comment]
We use an equivalent of `SmartContract.Helper.MaxVerificationGas` (not doubled, frankly I think it should be enough) as a limit in neo-go already.

**Comment by erikzhang** [comment]
@shargon I think it should be consistent with `VerifyWitnesses()`.

https://github.com/neo-project/neo/blob/d73d190c87eb999c36934c1fe93aaee8c818701a/src/neo/SmartContract/Helper.cs#L169

**Review by erikzhang — APPROVED** [review]

**Comment by shargon** [comment]
@erikzhang please check it again

**Review by erikzhang — APPROVED** [review]

File: src/neo/Wallets/Wallet.cs [hunk 1]
```diff
@@ -348,6 +348,8 @@ private Transaction MakeTransaction(DataCache snapshot, byte[] script, Signer[]
                 }
 
                 tx.NetworkFee = CalculateNetworkFee(snapshot, tx);
+                if (tx.NetworkFee - (tx.Size * NativeContract.Policy.GetFeePerByte(snapshot)) > MaxVerificationGas)
```
Comment (erikzhang) [inline src/neo/Wallets/Wallet.cs:351]: You can't calculate the size here because `Witnesses` is null.

**Review by erikzhang — CHANGES_REQUESTED** [review]

File: src/neo/Wallets/Wallet.cs (hunk 1 above)
Comment (shargon) [reply src/neo/Wallets/Wallet.cs]: Agree, we should do that in neo-node

**Review by erikzhang — APPROVED** [review]

**Pull request merged**